app/routes/circulars._archive._index/route.tsx

```diff
@@ -111,23 +112,36 @@ export async function action({ request }: ActionFunctionArgs) {
       if (!createdOnDate || !createdOn)
         throw new Response(null, { status: 400 })
 
-      const {
-        request: { id: zendeskTicketId },
-      } = await postZendeskRequest({
-        requester: { name: user.name, email: user.email },
-        subject: `Change Request for Circular ${circularId}`,
-        comment: {
-          body: `${user.name} has requested an edit. Review at ${origin}/circulars`,
-        },
-      })
+      let zendeskTicketId
+
+      try {
+        zendeskTicketId = (
+          await getChangeRequest(parseFloat(circularId), user.sub)
+        ).zendeskTicketId
+      } catch (error) {
+        console.info('No existing request found')
+      }
+
+      if (!zendeskTicketId) {
+        ;({
+          request: { id: zendeskTicketId },
+        } = await postZendeskRequest({
+          requester: { name: user.name, email: user.email },
+          subject: `Change Request for Circular ${circularId}`,
+          comment: {
+            body: `${user.name} has requested an edit. Review at ${origin}/circulars`,
+          },
+        }))
+        if (!zendeskTicketId) throw new Response(null, { status: 500 })
+      }
```

Comment:
Can `postZendeskRequest` even return without raising an exception if it fails to create a ticket?

Reply:
I'm not sure, this is more to resolve the type issue of it being either a number or undefined

Reply:
This is all it returns if it fails:

```js
  if (!response.ok) {
    console.error(response)
    throw new Error(`Request failed with status ${response.status}`)
  }
```

This check is just to resolve the type issue of it being potentially undefined

Reply:
Could we have `postZendeskRequest` raise an exception if it does not return a new ticket ID?